schematic/store/synapse.py

```diff
@@ -2130,50 +2214,45 @@ def upsertTable(synStore: SynapseStorage, sg: SchemaGenerator, tableToLoad: pd.D
         
 
         Args:
-            tableToLoad: manifest formatted appropriately for the table
-            tableName: name of the table to be uploaded
-            existingTableId: synId of the existing table to be replaced     
-            datasetId: synID of the dataset for the manifest    
-            columnTypeDict: dictionary schema for table columns: type, size, etc
+            sg: SchemaGenerator instance
             
-
         Returns:
            existingTableId: synID of the already existing table that had its metadata replaced
         """            
 
-        username, authtoken = TableOperations._get_schematic_db_creds(synStore)
+        username, authtoken = self._get_schematic_db_creds()
 
-        synConfig = SynapseConfig(username, authtoken, synStore.getDatasetProject(datasetId))
+        synConfig = SynapseConfig(username, authtoken, self.synStore.getDatasetProject(self.datasetId))
         synapseDB = SynapseDatabase(synConfig)
 
         try:
             # Try performing upsert
-            synapseDB.upsert_table_rows(table_name=tableName, data=tableToLoad)
+            synapseDB.upsert_table_rows(table_name=self.tableName, data=self.tableToLoad)
         except(SynapseHTTPError) as ex:
             # If error is raised because Table has old `Uuid` column and not new `Id` column, then handle and re-attempt upload
             if 'Id is not a valid column name or id' in str(ex):
-                TableOperations._update_table_uuid_column(synStore, existingTableId, sg)
-                synapseDB.upsert_table_rows(table_name=tableName, data=tableToLoad)
+                self._update_table_uuid_column(sg)
+                synapseDB.upsert_table_rows(table_name=self.tableName, data=self.tableToLoad)
             # Raise if other error
             else:
                 raise ex
 
-        return existingTableId
+        return self.existingTableId
 
-    def _update_table_uuid_column(synStore: SynapseStorage, table_id: str, sg: SchemaGenerator,) -> None:
```

Comment:
@thomasyu888 here's the relevant chunk of code for renaming a column in an existing table

Reply:
@thomasyu888 for sure, hope you find it useful
For this chuck of code, right now I don't plan for this to be reused. When speaking with @andrewelamb he mentioned that this isn't something we want to be doing often, in the context of treating Synapse tables like a database. The preferred method would be for users to rebuild their schema and reupload if they need to make a change to a column. This chunk enables backwards compatibility because of a change on our end so that users wouldn't have to manually do it themselves for this special case.

Reply:
I definitely find this helpful!  That makes sense to me - Synapse Tables is definitely not a database.  In terms of backwards compatibility - I leave it to you all to decide whats best.  What I look for code is

1. Reusability
2. Readabilty
3. Testability

I challenge everyone to think broader.  Yes, this particular code may not be re-used here, but table column renames is probably something that is helpful to the Sage community.  By pulling that core functionality out, it would be easier to test, and you could take that directly and create a PR into the python client.